text/0000-pack-toml.md

```diff
@@ -0,0 +1,123 @@
+# Meta
+[meta]: #meta
+- Name: Support for `pack.toml`
+- Start Date: 2021-11-16
+- Author(s): [jkutner](http://github.com/jkutner)
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal for a `pack.toml` file that is a special case of the `project.toml` descriptor file. The `pack.toml` will include configuration options specific to the Pack CLI, which may not be honored by other platforms.
+
+# Definitions
+[definitions]: #definitions
+
+- *project descriptor* - a extension spec defining the [`project.toml`](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md) file.
+
+
+# Motivation
+[motivation]: #motivation
+
+The `project.toml` file has been a useful and essential tool for buildpack users who need to codify certain arguments of their builds. However, there has been much debate around how to handle configuration options that may not be supported by all platforms.
+
+We desire a descriptor file that is:
+* specific to pack (i.e. not platform-neutral)
+* advertises that the repo containing the file can be built with pack
+* adheres to the best practices and learnings from [`project.toml`](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md)
+
+# What it is
+[what-it-is]: #what-it-is
+
+The Pack CLI will support a new descriptor file, `pack.toml`, that is a superset of the project descriptor.
+* An app MAY have a `pack.toml`
+* An app MAY have a `project.toml`
+* An app MAY have both a `pack.toml` and a `projec.toml`, in which case they will be merged.
+
+The `pack.toml` will support an additional `[io.buildpacks.pack]` table that is NOT defined in the [project descriptor spec](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md). The `[io.buildpacks.pack]` table will have the following schema:
+
+```toml
+[io.buildpacks.pack]
+builder = "<string (optional)>"
+```
+
+Where:
+
+* `builder` is a uri to a CNB Builder docker image
+
+## Example
+
+```
+[io.buildpacks.build]
+exclude = ["/README.md"]
+
+[[io.buildpacks.group]]
+id = "example/java"
+version = "1.0.0"
+
+[io.buildpacks.pack]
+builder = "example/my-builder-image"
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The Pack CLI will honor both the `pack.toml` and `project.toml` file descriptors with the following rules:
+
+* If one or more descriptors are provided on the command line they will all be merged and used.
```

Comment:
I wonder if this would be simpler if we only had one flag, say something like `--config` that could accept either `pack.toml` OR `project.toml`.

The same could then apply to an inferred discovery mechanism where if the `--config` flag is not presented then it's looked up by precedence and NOT merged.

1. Look up `pack.toml`
2. Look up `project.toml`